Sources/Parameter.swift

```diff
@@ -1,61 +1,106 @@
+        case location = "in"
     }
-
+    
     init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: CodingKeys.self)
-        let fixedFields = try FixedParameterFieldsBuilder(from: decoder)
-        switch fixedFields.location {
-        case .body:
-            let schema = try values.decode(SchemaBuilder.self, forKey: .schema)
-            self = .body(fixedFieldsBuilder: fixedFields, schemaBuilder: schema)
-        case .query, .header, .path, .formData:
-            self = .other(fixedFieldsBuilder: fixedFields, itemsBuilder: try ItemsBuilder(from: decoder))
-        }
-    }
-
-    func encode(to encoder: Encoder) throws {
-        switch self {
-        case .body(let fixedFieldsBuilder, let schemaBuilder):
-            var container = encoder.container(keyedBy: CodingKeys.self)
-            try fixedFieldsBuilder.encode(to: encoder)
-            try container.encode(schemaBuilder, forKey: .schema)
-        case .other(let fixedFieldsBuilder, let itemsBuilder):
-            try fixedFieldsBuilder.encode(to: encoder)
-            try itemsBuilder.encode(to: encoder)
-        }
+        self.name = try values.decode(String.self, forKey: .name)
+        self.location = try values.decode(ParameterLocation.self, forKey: .location)
+        self.definitionBuilder = try ParameterDefinitionBuilder(from: decoder)
     }
 }
 
 extension ParameterBuilder: Builder {
     typealias Building = Parameter
-
+    
     func build(_ swagger: SwaggerBuilder) throws -> Parameter {
-        switch self {
-        case .body(let fixedFieldsBuilder, let schemaBuilder):
-            return .body(fixedFields: try fixedFieldsBuilder.build(swagger),
-                         schema: try schemaBuilder.build(swagger))
-        case .other(let fixedFieldsBuilder, let itemsBuilder):
-            return .other(fixedFields: try fixedFieldsBuilder.build(swagger),
-                          items: try itemsBuilder.build(swagger))
-        }
+        let definition = try self.definitionBuilder.build(swagger)
+        return Parameter(name: self.name,
+                         location: self.location,
+                         description: definition.description,
+                         required: definition.required,
+                         deprecated: definition.deprecated,
+                         allowEmptyValue: definition.allowEmptyValue,
+                         style: definition.style,
+                         explode: definition.explode,
+                         allowReserved: definition.allowReserved,
+                         schema: definition.schema,
+                         example: definition.example,
+                         examples: definition.examples,
+                         content: definition.content)
     }
 }
 
+
 extension ParameterBuilder {
     static func resolve(_ swagger: SwaggerBuilder, reference: Reference<ParameterBuilder>) throws
         -> Either<Parameter, Structure<Parameter>>
```

Comment:
### [Parameter Object](https://swagger.io/specification/#parameterObject)

As per new OAS3 Location values - body is now excluded (handled via separate Request Body Object).